### PR TITLE
Implement GET /profile, DELETE /logout

### PR DESCRIPTION
**Pull request by dypher91**

### CODE REVIEW

File: GiTinder/Controllers/UsersController.cs [hunk 1]
```diff
@@ -45,5 +45,55 @@ public async Task<GeneralApiResponseBody> Login([FromBody] LoginRequestBody logi
...
+            var usertoken = Request.Headers["X-Gitinder-Token"];
+            GeneralApiResponseBody responseBody;
+
+            if (string.IsNullOrEmpty(usertoken))
+            {
+                Response.StatusCode = 403;
+                responseBody = new ErrorResponseBody("Unauthorized request!");
+            }
+            else if (_userServices.TokenExists(usertoken))
+            {
```
Comment (adamgyulavari) [inline GiTinder/Controllers/UsersController.cs:74]: @dypher91 you should actually remove the token from the user in the database

File: GiTinder/Controllers/UsersController.cs [hunk 2]
```diff
@@ -45,5 +45,55 @@ public async Task<GeneralApiResponseBody> Login([FromBody] LoginRequestBody logi
...
+            var responseProfile = _userServices.FindUserByUserToken(usertoken);
+
+            if (string.IsNullOrEmpty(usertoken))
+            {
+                Response.StatusCode = 403;
+                responseBody = new ErrorResponseBody("Unauthorized request!");
+            }
+            if (_userServices.TokenExists(usertoken))
+            {
+                responseBody = new ProfileResponse(responseProfile.Username, responseProfile.Avatar,responseProfile.Repos);
```
Comment (adamgyulavari) [inline GiTinder/Controllers/UsersController.cs]: @dypher91 repos should be a list of string splitted by the semicolon

File: GiTinder/Models/Responses/ProfileResponse.cs [hunk 3]
```diff
@@ -7,24 +7,31 @@
...
         public string Repos { get; set; }
+
         [JsonProperty("languages")]
         public string Languages { get; set; }
 
+        [JsonProperty("snippets")]
+        public string Snippets { get; set; }
 
-        public ProfileResponse()
+        public ProfileResponse(string userName, string uRl, string rePos)
```
Comment (adamgyulavari) [inline GiTinder/Models/Responses/ProfileResponse.cs]: @dypher91 `username`, `url`, `repos`

File: GiTinder/Models/Responses/ProfileResponse.cs [hunk 4]
```diff
@@ -7,24 +7,31 @@
...
-            Avatar = "https://avatars1.githubusercontent.com/u/5855091?s=400&v=4";
-            Repos = "string";
-            Languages = "eng";
-
+            Status = "ok";
+            Username = userName;
+            Url = uRl;
+            Repos = rePos;
+            Languages = "Java";
+            Snippets = "https://github.com/adamgyulavari/gf-chatapp";
```
Comment (adamgyulavari) [inline GiTinder/Models/Responses/ProfileResponse.cs]: @dypher91 these 3 should be `list<string>`

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: GiTinder/Models/Responses/ProfileResponse.cs (hunk 3 above)
Comment (dypher91) [reply GiTinder/Models/Responses/ProfileResponse.cs]: done

File: GiTinder/Controllers/UsersController.cs (hunk 1 above)
Comment (dypher91) [reply GiTinder/Controllers/UsersController.cs:74]: done

File: GiTinder/Models/Responses/ProfileResponse.cs (hunk 4 above)
Comment (dypher91) [reply GiTinder/Models/Responses/ProfileResponse.cs]: done

File: GiTinder/Controllers/UsersController.cs (hunk 2 above)
Comment (dypher91) [reply GiTinder/Controllers/UsersController.cs]: done

**Comment by adamgyulavari** [comment]
@dypher91 resolve conflicts!

**Pull request merged**